servo-media/src/lib.rs

```diff
@@ -1,152 +1,57 @@
 pub extern crate servo_media_audio as audio;
-#[cfg(any(
-    all(target_os = "android", target_arch = "arm"),
-    target_arch = "x86_64"
-))]
-extern crate servo_media_gstreamer;
+
 pub extern crate servo_media_player as player;
 pub extern crate servo_media_webrtc as webrtc;
 pub extern crate servo_media_streams as streams;
-use std::any::Any;
+use std::ops::Deref;
 use std::sync::{self, Arc, Mutex, Once};
 
 use audio::context::{AudioContext, AudioContextOptions};
-use audio::decoder::DummyAudioDecoder;
-use audio::sink::{AudioSinkError, DummyAudioSink};
-use audio::AudioBackend;
-use player::{DummyPlayer, Player, PlayerBackend};
-use streams::{DummyMediaOutput, MediaStream, MediaOutput};
+use player::Player;
+use streams::{MediaStream, MediaOutput};
 use streams::capture::MediaTrackConstraintSet;
-use webrtc::{DummyWebRtcController, WebRtcBackend};
 use webrtc::{WebRtcController, WebRtcSignaller};
 
-pub struct ServoMedia;
+pub struct ServoMedia(Box<Backend>);
 
 static INITIALIZER: Once = sync::ONCE_INIT;
 static mut INSTANCE: *mut Mutex<Option<Arc<ServoMedia>>> = 0 as *mut _;
 
-pub struct DummyMediaStream;
-impl MediaStream for DummyMediaStream {
-    fn as_any(&self) -> &Any {
-        self
-    }
-    fn as_mut_any(&mut self) -> &mut Any {
-        self
-    }
-}
-
-pub struct DummyBackend {}
-
-impl AudioBackend for DummyBackend {
-    type Decoder = DummyAudioDecoder;
-    type Sink = DummyAudioSink;
-    fn make_decoder() -> Self::Decoder {
-        DummyAudioDecoder
-    }
-
-    fn make_sink() -> Result<Self::Sink, AudioSinkError> {
-        Ok(DummyAudioSink)
-    }
-}
-
-impl PlayerBackend for DummyBackend {
-    type Player = DummyPlayer;
-    fn make_player() -> Self::Player {
-        DummyPlayer {}
-    }
+pub trait BackendInit {
+    fn init() -> Box<Backend>;
 }
 
-impl WebRtcBackend for DummyBackend {
-    type Controller = DummyWebRtcController;
-    fn construct_webrtc_controller(
-        _: Box<WebRtcSignaller>,
-        _: WebRtcController,
-    ) -> Self::Controller {
-        DummyWebRtcController
-    }
+pub trait Backend: Send + Sync {
+    fn make_player(&self) -> Box<Player>;
```

Comment:
uber-nit: s/make_player/create_player/g as all the other methods are `create_*`.